src/common/transformations/src/transformations/op_conversions/convert_gather_to_compressed.cpp

```diff
@@ -146,3 +147,46 @@ ov::pass::ConvertGatherToGatherCompressed::ConvertGatherToGatherCompressed() {
     auto m = std::make_shared<ov::pass::pattern::Matcher>(gather_m, "ConvertGatherToGatherCompressed");
     this->register_matcher(m, callback);
 }
+
+ov::pass::MoveDecompressionAfterGather::MoveDecompressionAfterGather() {
+    using namespace ov::pass::pattern;
+
+    auto compressed_constant_without_scale = [](const ov::Output<ov::Node>& output) {
+        return output.get_element_type() == ov::element::f16 || output.get_element_type() == ov::element::bf16;
+    };
+    auto dicts_wo_scale = wrap_type<ov::op::v0::Constant>(compressed_constant_without_scale);
+    auto convert_wo_scale = wrap_type<ov::op::v0::Convert>({dicts_wo_scale}, pattern::consumers_count(1));
+    auto gather_wo_scale =
+        wrap_type<ov::op::v8::Gather>({convert_wo_scale, any_input(), wrap_type<ov::op::v0::Constant>()});
+
+    ov::matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto constant_node = pattern_map.at(dicts_wo_scale).get_node_shared_ptr();
+        auto convert_node = pattern_map.at(convert_wo_scale).get_node_shared_ptr();
+        auto gather_node = pattern_map.at(gather_wo_scale).get_node_shared_ptr();
+        if (transformation_callback(gather_node)) {
+            return false;
+        }
+
+        if (convert_node->get_output_target_inputs(0).size() != 1 ||
+            convert_node->get_output_element_type(0) != element::Type_t::f32) {
+            return false;
+        }
```

Comment:
1. Consumers count is already checked in the matcher
2. Output precision check should be also moved in the matcher: let's create a custom predicate which will call `pattern::consumers_count(1)(output) && pattern::type_matches(ov::element::f32)(output)`

Reply:
1. removed double check and 2. moved convert output precision f32 to matcher, thanks Vladislav!